### DISCUSSION TITLE
F3 in Kitty keyboard protocol

### DISCUSSION THREAD

**Discussion by djarb** [kovidgoyal/kitty#5813] [Q&A] [answered]
kitty +kitten show_key -m kitty shows Shift+F3 as producing
CSI 57366 ; 2 u

Which is great, but printf '\x1b[>11u'; cat -v shows Shift+F3 as producing
^[[1;2R

The latter is not great. In fact, that's a Report Cursor Position (CPR) response. Do not want.

How can I instruct Kitty (and other protocol-compatible terminals) to send the CSI u version of the event? I tried the obvious, and dug into the kitten to replicate the same initialization escape codes it uses, to no avail. And now, having looked into the Kitty source code that generates event codes, I don't even see how it's possible for it to generate the CSI u code that show_key is reporting.

Anybody got a clue for me?

**Comment by kovidgoyal** [comment, marked as answer]
That's a bug in show_key already fixed in master. 
And yes, there is a conflict with CPR for press events which I didn't realize since I never
use CPR. I will change kitty to produce CSI 13 ~ instead for F3 which is
allowed by the spec.

On Fri, Dec 23, 2022 at 03:58:09AM -0800, djarb wrote:
> kitty +kitten show_key -m kitty shows Shift+F3 as producing
> CSI 57366 ; 2 u
> 
> Which is great, but printf '\x1b[>11u'; cat -v shows Shift+F3 as producing
> ^[[1;2R
> 
> The latter is not great. In fact, that's a Report Cursor Position (CPR) response. Do not want.
> 
> How can I instruct Kitty (and other protocol-compatible terminals) to send the CSI u version of the event? I tried the obvious, and dug into the kitten to replicate the same initialization escape codes it uses, to no avail. And now, having looked into the Kitty source code that generates event codes, I don't even see how it's possible for it to generate the CSI u code that show_key is reporting.
> 
> Anybody got a clue for me?
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/5813
> You are receiving this because you are subscribed to this thread.
> 
> Message ID: ***@***.***>

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Comment by kovidgoyal** [comment]
https://github.com/kovidgoyal/kitty/commit/cd92d50a0d34b7c3525377770fba47ce75cf1cfc
